src/MediatR/Internal/NotificationHandlerWrapper.cs

```diff
@@ -18,7 +18,7 @@ public override Task Handle(INotification notification, CancellationToken cancel
         {
             var handlers = serviceFactory
                 .GetInstances<INotificationHandler<TNotification>>()
-                .Select(x => x.Handle((TNotification)notification, cancellationToken));
+                .Select(async x => await x.Handle((TNotification)notification, cancellationToken));
```

Comment:
Should this await have `.ConfigureAwait(false)`?

Reply:
if it is in PublishCore doesn't that work?  If it is here that can't be overridden (not sure if that matters)